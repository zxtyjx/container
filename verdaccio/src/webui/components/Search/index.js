
import React from 'react';
import PropTypes from 'prop-types';

import classes from './search.scss';

const noSubmit = (e) => {
  e.preventDefault();
};

const Search = (props) => {
    return (
      <form autoComplete="off" onSubmit={noSubmit}>
        <input
          name="search-box"
          type="text"
          placeholder={props.placeHolder}
          className={classes.searchBox}
          onChange={props.handleSearchInput}
          autoComplete="off"
        />
      </form>
    );
};

Search.defaultProps = {
  placeHolder: 'Type to search...'
};

Search.propTypes = {
  handleSearchInput: PropTypes.func.isRequired,
  placeHolder: PropTypes.string,
};

export default Search;
