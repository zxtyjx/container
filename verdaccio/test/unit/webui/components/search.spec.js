/**
 * Search component
 */

import React from 'react';
import { shallow } from 'enzyme';
import Search from '../../../../src/webui/components/Search/index';
console.error = jest.fn();

describe('<Search /> component', () => {
  it('should give error for the required fields', () => {
    const wrapper = shallow(<Search />);
    expect(console.error).toBeCalled();
    expect(wrapper.find('input').prop('placeholder')).toEqual(
      'Type to search...'
    );
  });

  it('should have <input /> element with correct properties', () => {
    const props = {
      handleSearchInput: () => {},
      placeHolder: 'Test placeholder'
    };
    const wrapper = shallow(<Search {...props} />);
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('input').prop('placeholder')).toEqual(
      'Test placeholder'
    );
  });

  it('should call the handleSearchInput function', () => {
    const props = {
      handleSearchInput: jest.fn()
    };
    const wrapper = shallow(<Search {...props} />);
    wrapper.find('input').simulate('change');
    expect(props.handleSearchInput).toBeCalled();
  });

  it('should match the snapshot', () => {
    const props = { handleSearchInput: () => {} };
    const wrapper = shallow(<Search {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
