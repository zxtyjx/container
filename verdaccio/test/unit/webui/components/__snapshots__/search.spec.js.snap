// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should match the snapshot 1`] = `"<form autoComplete=\\"off\\"><input type=\\"text\\" name=\\"search-box\\" placeholder=\\"Type to search...\\" class=\\"searchBox\\" autoComplete=\\"off\\"/></form>"`;
